frontend/src/pages/store/index.svelte

```diff
@@ -1,27 +1,209 @@
 
 	metatags.title = 'Stores - Kitcoin';
 
 	let ctx = getContext('userInfo');
 	let authMsg = null;
+	let userInfo;
 	(async () => {
-		let info = (await ctx) || null;
-		if (!info) return (authMsg = 'NO_USER');
+		userInfo = (await ctx) || null;
+		if (!userInfo) return (authMsg = 'NO_USER');
 		if (
-			!info.scopes.includes(
+			!userInfo.scopes.includes(
 				'https://www.googleapis.com/auth/classroom.courses.readonly',
 			)
 		)
 			authMsg = 'CLASSROOM';
 	})();
+
+	let stores = undefined;
+	async function load(useCache) {
+		await getStores(useCache)
+			.then(x => {
+				stores = x;
+			})
+			.catch(e => {
+				stores = null;
+			});
+		return;
+	}
+	load();
+
+	// Manage items
+	let manageFormData = {
+		isValid: false,
+		values: {},
+		errors: {},
+	};
+	let manageForm;
+	let manageValidators = {
+		name: e => {
+			let v = e.value;
+			if (!v) return e && e.type == 'blur' ? 'Name is required' : '';
+			return null;
+		},
+		description: e => {
+			let v = e.value;
+			return null;
+		},
+		classIDs: e => {
+			let v = e.value;
+			return null;
+		},
+		public: e => {
+			let v = e.value;
+			return null;
+		},
+		classes: e => {
+			let v = e.value?.value;
+			return null;
+		},
+		managers: e => {
+			let v = e.value?.value;
+			return null;
+		},
+		users: e => {
+			let v = e.value?.value;
+			return null;
+		},
+	};
+
+	let modalStore = null;
+	let modalOpen = false;
+	let submitStatus = null;
+	let resetTimeout;
+
+	async function setModalStore(store) {
+		submitStatus = null;
+		let classes = store ? await getClasses('teacher') : null;
+		modalStore = store;
+		manageForm.reset();
+		if (store) {
+			manageForm.values.name = modalStore.name;
+			manageForm.values.description = modalStore.description;
+			manageForm.values.classes = modalStore.classIDs.map(x => {
+				let classData = classes.find(c => c.id == x);
+
+				return classData
+					? {
+							text: classData.name,
+							value: classData.id,
+					  }
+					: {
+							text: `Unknown Class (${c.id})`,
+							value: c.id,
+					  };
+			});
+			manageForm.values.public = modalStore.public;
+			manageForm.values.managers = modalStore.managers.map(x => ({
+				text: x.name,
+				value: x.id,
+			}));
+			manageForm.values.users = modalStore.users.map(x => ({
+				text: x.name,
+				value: x.id,
+			}));
+		}
+		modalOpen = true;
+	}
+
+	async function doManageStore(e) {
+		e.preventDefault();
+		if (!manageFormData.isValid) return false;
+		submitStatus = 'LOADING';
+		let res = await fetch(
+			modalStore ? `/api/store/${modalStore._id}` : `/api/stores`,
+			{
+				method: modalStore ? 'PATCH' : 'POST',
+				headers: {
+					'Content-Type': 'application/json',
+				},
+				body: JSON.stringify({
+					name: manageFormData.values.name,
+					description: manageFormData.values.description || null,
+					classIDs: (manageFormData.values.classes || []).map(
+						x => x.value,
+					),
+					public: manageFormData.values.public ?? modalStore.public,
```

Comment:
TLDR; users with `STAFF` but not `ADMIN` cannot create stores.

For `STAFF` users, `manageFormData.values.public` is undefined because on line 368 you only give `ADMIN` users access to a _public_ toggle and the value does not have a default. Additionally, when creating a store, `modalStore` is undefined as well, so an error is thrown when `modalStore.public` can't be read. This means that the request is never sent and the store is never created. So pretty much staff can't create stores.  
If this is intended behavior then staff shouldn't be able to open the store creation popup in the first place.